File: BubblerApp/BubblerApp/Models/ReportReason.swift
import Foundation

/// Cap for untrusted reporter notes; matches backend `DETAILS_MAX_LENGTH`.
enum ReportDetailsLimits {
    static let maxLength = 2000
}

/// Why a user is reporting another person's post.
/// Matches Phase 0 hard-removal categories plus spam / other.
/// `illegalContent` is the severe-illegal / CSAM isolation bucket (escalation: L11).
enum ReportReason: String, CaseIterable, Identifiable {
    case illegalContent
    case severeViolence
    case nonConsensualSexualContent
    case harassment
    case spam
    case other

    var id: String { rawValue }

    /// API / DB reason string (`snake_case`).
    var apiValue: String {
        switch self {
        case .illegalContent: "illegal_content"
        case .severeViolence: "severe_violence"
        case .nonConsensualSexualContent: "non_consensual_sexual_content"
        case .harassment: "harassment"
        case .spam: "spam"
        case .other: "other"
        }
    }

    var title: String {
        switch self {
        case .illegalContent: "Illegal content"
        case .severeViolence: "Severe violence"
        case .nonConsensualSexualContent: "Non-consensual sexual content"
        case .harassment: "Harassment or threats"
        case .spam: "Spam"
        case .other: "Other"
        }
    }
}

File: BubblerAndroid/app/src/main/java/com/bubbler/android/data/model/ReportReason.kt
package com.bubbler.android.data.model

/**
 * Why a user is reporting another person's post.
 * Matches Phase 0 hard-removal categories plus spam / other.
 * [ILLEGAL_CONTENT] is the severe-illegal / CSAM isolation bucket (escalation: L11).
 */
enum class ReportReason(val title: String, val apiValue: String) {
    ILLEGAL_CONTENT("Illegal content", "illegal_content"),
    SEVERE_VIOLENCE("Severe violence", "severe_violence"),
    NON_CONSENSUAL_SEXUAL_CONTENT("Non-consensual sexual content", "non_consensual_sexual_content"),
    HARASSMENT("Harassment or threats", "harassment"),
    SPAM("Spam", "spam"),
    OTHER("Other", "other"),
}

/** Cap for untrusted reporter notes; matches backend `DETAILS_MAX_LENGTH`. */
object ReportDetailsLimits {
    const val MAX_LENGTH = 2000
}
